Examples/ObjectiveC/LightExample.m

```diff
@@ -0,0 +1,100 @@
+//
+//  LightExample.m
+//  Examples
+//
+//  Created by Jordan Kiley on 6/22/17.
+//  Copyright © 2017 Mapbox. All rights reserved.
+//
+
+#import "LightExample.h"
+@import Mapbox;
+
+@interface LightExample () <MGLMapViewDelegate>
+
+@property (nonatomic) MGLMapView *mapView;
+@property (nonatomic) MGLLight *light;
+@property (nonatomic) UISlider *slider;
+
+@end
+
+NSString *const MBXExampleLight = @"LightExample";
+
+@implementation LightExample
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    // Set the map style to Mapbox Light Style version 9. The map's source will be queried later in this example.
+    self.mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds styleURL:[MGLStyle streetsStyleURLWithVersion:9]];
+    self.mapView.delegate = self;
+    
+    // Center the map on the Flatiron Building in New York, NY.
+    self.mapView.camera = [MGLMapCamera cameraLookingAtCenterCoordinate:CLLocationCoordinate2DMake(40.7411, -73.9897) fromDistance:600 pitch:45 heading:200];
+    self.mapView.tintColor = [UIColor grayColor];
+    
+    [self.view addSubview:self.mapView];
+    
+    [self addSlider];
+}
+
+// Add a slider to the map view. This will be used to adjust the map's light object.
+- (void)addSlider {
+    self.slider = [[UISlider alloc] initWithFrame:CGRectMake(self.view.frame.size.width / 8, self.view.frame.size.height - 60, self.view.frame.size.width * 0.75, 20)];
+    self.slider.autoresizingMask = UIViewAutoresizingFlexibleTopMargin | UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleRightMargin;
+    self.slider.minimumValue = -180;
+    self.slider.maximumValue = 180;
+    self.slider.value = 0;
+    [self.slider addTarget:self action:@selector(shiftLight) forControlEvents:UIControlEventValueChanged];
+    [self.view addSubview:self.slider];
+}
+
+- (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
+    
+    // Add a MGLFillExtrusionStyleLayer.
+    [self addFillExtrusionLayer:style];
+    
+    // Create an MGLLight object.
+    self.light = [[MGLLight alloc] init];
+    
+    // Create an MGLSphericalPosition and set the radial, azimuthal, and polar values.
+    // Radial : Distance from the center of the base of an object to its light.
+    // Azimuthal : Position of the light relative to its anchor.
+    // Polar : The height of the light.
+    MGLSphericalPosition position = MGLSphericalPositionMake(5, 180, 80);
+    self.light.position = [MGLStyleValue valueWithRawValue:[NSValue valueWithMGLSphericalPosition:position]];
+    
+    // Set the light anchor to the map and add the light object to the map view's style. The light anchor can be the viewport (or rotates with the viewport) or the map (rotates with the map).
+    // self.light.anchor = [MGLStyleValue valueWithRawValue:[NSValue valueWithMGLLightAnchor:MGLLightAnchorViewport]];
```

Comment:
Ah, I mistook this for accidental leftover code — do you think `MGLLightAnchorViewport` could be worked into the comment instead?